tox.ini

```diff
@@ -1,21 +1,59 @@
-# spell-checker:ignore toxinidir passenv usedevelop testenv basepython envpython linkcheck changedir envdir envlist minversion toxworkdir posargs envname
+# spell-checker:ignore toxinidir passenv usedevelop testenv basepython envpython linkcheck changedir envdir envlist minversion toxworkdir posargs envname skipsdist setenv
 [tox]
 envlist =
   lint
   docs
+  test
 minversion = 3.21.0
 skip_install = true
-
+skipsdist = true
 
 [docs]
 # do not use continuation with vars, see https://github.com/tox-dev/tox/issues/2069
 sphinx_common_args =
   -j auto {tty:--color} -a -n -W --keep-going -T -d "{temp_dir}/.doctrees" . "{envdir}/docs_out"
 
 [testenv]
+description =
+  Run tests
+  test-node12: with node 12
+  test-node14: with node 14
+  test-node16: with node 16
 passenv =
   CI
   GITHUB_ACTIONS
+  HOME
+  SSH_AUTH_SOCK
+  TERM
+setenv =
+  FORCE_COLOR=1
+  NODE_VERSION=stable
+  node12: NODE_VERSION=12
+  node14: NODE_VERSION=14
+  node16: NODE_VERSION=16
+allowlist_externals =
+  bash
+  node
+# Activation of node version is not persistent and nvm works only from
+# within `bash --login`, do not call nvm/npm/node without it under tox as
+# you might end-up using system version, if any
+commands_pre =
+  {node12,node14,node16}: bash --login -c "nvm install --latest-npm --default {env:NODE_VERSION}"
+commands_post =
+  # restore nvm default alias to stable version
+  {node12,node14,node16}: bash --login -c "nvm alias default stable"
```

Comment:
Having added this into the base toxenv, don't forget to reset these in every toxenv that inherits this and doesn't need this garbage. I'm pretty sure this breaks bits of the the release automation, for example.